src/test/java/cloud/fogbow/common/plugins/cloudidp/openstack/v3/OpenStackIdentityProviderPluginTest.java

```diff
@@ -0,0 +1,162 @@
+    @Test(expected = UnexpectedException.class) // verify
+    public void testGetCloudUserWithEmptyCredentialsMap() throws FogbowException {
+        // set up
+        Map<String, String> credentials = new HashMap<String, String>();
+
+        // exercise
+        this.plugin.getCloudUser(credentials);
+    }
+
+    // test case: When invoking the getCloudUser method with a credential map
+    // without a user name, it must throw an UnexpectedException.
+    @Test(expected = UnexpectedException.class) // verify
+    public void testGetCloudUserWithoutUsername() throws FogbowException {
+        // set up
+        Map<String, String> credentials = createCredentials(null, ANY_VALUE, ANY_VALUE, ANY_VALUE);
+
+        // exercise
+        this.plugin.getCloudUser(credentials);
+    }
+
+    // test case: When invoking the getCloudUser method with a credential map
+    // without a password, it must throw an UnexpectedException.
+    @Test(expected = UnexpectedException.class) // verify
+    public void testGetCloudUserWithoutPassword() throws FogbowException {
+        // set up
+        Map<String, String> credentials = createCredentials(ANY_VALUE, null, ANY_VALUE, ANY_VALUE);
+
+        // exercise
+        this.plugin.getCloudUser(credentials);
+    }
+
+    // test case: When invoking the getCloudUser method with a credential map
+    // without a domain, it must throw an UnexpectedException.
+    @Test(expected = UnexpectedException.class) // verify
+    public void testGetCloudUserWithoutDomain() throws FogbowException {
+        // set up
+        Map<String, String> credentials = createCredentials(ANY_VALUE, ANY_VALUE, null, ANY_VALUE);
+
+        // exercise
+        this.plugin.getCloudUser(credentials);
+    }
+
+    // test case: When invoking the getCloudUser method with a credential map
+    // without a domain, it must throw an UnexpectedException.
+    @Test(expected = UnexpectedException.class) // verify
+    public void testGetCloudUserWithoutProject() throws FogbowException {
+        // set up
+        Map<String, String> credentials = createCredentials(ANY_VALUE, ANY_VALUE, ANY_VALUE);
+
+        // exercise
+        this.plugin.getCloudUser(credentials);
+    }
+
+    // test case: When invoking the getCloudUser method with a credential map
+    // with correct data, without providing projectId
+    @Test
+    public void testGetCloudUserUnscoped() throws FogbowException {
+        // setup
+        OpenStackV3User fakeToken = new OpenStackV3User(ANY_VALUE, ANY_VALUE, ANY_VALUE, null);
+        mockServices(fakeToken);
+        Map<String, String> credentials = createCredentials(ANY_VALUE, ANY_VALUE, ANY_VALUE);
+
+        // exercise
+        this.plugin.getCloudUser(credentials);
+
+        // verify
+        Mockito.verify(this.plugin).getUnscopedCloudUserFromJson(Mockito.any());
+    }
+
+    // test case: When invoking the getCloudUser method with a credential map
+    // with correct data, providing projectId
```

Comment:
this description misses the expected condition. Note that, it requires this pattern "When X then Y"

Reply:
Done.